lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -303,6 +302,16 @@ void getGuesses(std::vector<Inst *> &Guesses,
         if (K == Inst::Sub && V2->SynthesisConstID != 0)
           continue;
 
+        // PRUNE: don't synthesize "{shl, ashr, lshr} x, x" as its results in poison
```

Comment:
comment isn't accurate, this operation results in either 0 or poison

Reply:
fixed it now